sklearn/decomposition/_dict_learning.py

```diff
@@ -909,6 +909,9 @@ def _transform(self, X, dictionary):
         SparseCoder."""
         X = self._validate_data(X, reset=False)
 
+        if hasattr(self, "alpha") and self.transform_alpha is None:
+            self.transform_alpha = self.alpha
```

Comment:
I think it deserves a comment explaining why we have to put it here because it's not intuitive.